### ISSUE TITLE
Pager artifacting on Android 4.2.2 and below

### ISSUE THREAD

**Issue by joshfriend** [prolificinteractive/material-calendarview#84]
Nexus 4 with 4.2.2 and Galaxy Nexus with 4.0.4 are both exhibiting this behavior:

![](http://i.imgur.com/vAduHW9.png)

There is no fade in from the side animations like on 4.4 and above, the view clips at the bounds of the current month view.

I've tried [this](http://stackoverflow.com/a/13910364/1255482) and also calling `invalidate` on the calendar when the month changes, but no luck.

**Comment by kulio** [comment]
I have the exact same error.

**Comment by dandc87** [comment]
I haven't been able to replicate the artifacts on the couple of test devices I have, but I think the offending lines might be one of these:
```java
root.setClipChildren(false);
root.setClipToPadding(false);
```
Could you remove them and see that fixes this?

**Comment by kulio** [comment]
Still not working for me

**Comment by joshfriend** [comment]
@dandc87 removing those lines fixed the issue on both devices. I did some more digging and think it might be related to [this](https://code.google.com/p/android/issues/detail?id=36788).

I added the following, which seems to fix the issue and allow newer devices to maintain the nice page fade on scroll:

```java
if (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT) {
    root.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
}
root.setClipChildren(false);
root.setClipToPadding(false);
```

**Comment by kulio** [comment]
Yes, it works, thank you too much.

Instead of  root.setLayerType(View.LAYER_TYPE_SOFTWARE, null); i used root.setLayerType(View.LAYER_TYPE_HARDWARE, null); because it seems to run better.

**Comment by joshfriend** [comment]
Thats odd because the threads on the android-dev mailing list I was reading indicated that the issue was caused by using hardware rendering.

**Comment by dandc87** [comment]
Hey @joshfriend or @kulio, I've made some big changes with #99 that should solve this issue. Either by fix it, or by allowing the implementer to `setClipToPadding()` and `setClipChildren()` on the MaterialCalendarView itself to fix it. Could either of you test the latest version from the snapshot repo?

**Comment by joshfriend** [comment]
I checked out the branch from #99 (gradle couldn't find `com.prolificinteractive:material-calendarview:0.8.0-SNAPSHOT`) and the issue isn't fixed automatically. I still had to do this to get it working right:

```java
if (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT) {
    widget.setClipToPadding(true);
    if (Build.VERSION.SDK_INT > Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
        widget.setClipChildren(true);
    }
}
```

Tested with these devices that previously exhibited the artifacting:

* 4.0.4 Galaxy Nexus
* 4.1.1 Nexus 4 (Emulator)
* 4.2.2 Nexus 4

KitKat and newer devices still work fine. Is there a reason not to do the above clip settings in the library itself?

Also, the swiping was noticeably slower on the older Galaxy Nexus device.
